docs/advanced_topics.rst

```diff
@@ -1,114 +1,253 @@
 
-DALI uses intermediate buffers to pass data between operators in the processing graph. With DALI, the memory is never freed but just enlarged when present buffers are not sufficient to hold the data. However, in some cases, even this limited number of allocations still could affect DALI performance. Hence, if the user knows how much memory each operator buffer needs, then it is possible to provide a hint to presize buffers before the first run.
+When one value is provided, it is used for all output buffers for an operator. When a list is
+provided, each buffer is presized to the corresponding size.
+To determine the amount of memory output that each operator needs, complete the following tasks:
 
-Two parameters are available: First, the ``bytes_per_sample`` pipeline argument, which accepts one value that is used globally across all operators for all buffers.
-The second parameter is the ``bytes_per_sample_hint`` per operator argument, which accepts one value or a list of values. When one value is provided it is used for all output buffers for a given operator. When a list is provided then each buffer is presized to the corresponding size.
+1) you might Create the pipeline by setting ``enable_memory_stats`` to True.
+2) Query the pipeline for the operator’s output memory statistics by calling the ``executor_meta``
+   method on the pipeline.
 
-To learn how much memory outputs of each operator need, the user may create the pipeline with ``enable_memory_stats`` set to ``True`` and then query the pipeline for the operator's output memory statistics by calling ``executor_meta`` method on the pipeline. The ``max_real_memory_size`` value tells what is the biggest tensor in the batch for the outputs that allocate memory per sample, not for the whole batch at the time, or average tensor size when the allocation is contiguous. Usually this value is the one that should be provided to ``bytes_per_sample_hint``.
+The ``max_real_memory_size`` value represents the biggest tensor in the batch for the outputs that
+allocate memory per sample and not for the entire batch at the time or the average tensor size when
+the allocation is contiguous. This value should be provided to ``bytes_per_sample_hint``.
 
-Prefetching queue depth
+Prefetching Queue Depth
 -----------------------
 
-The purpose of this functionality is to average the processing time between batches when the variation from batch to batch is high.
-DALI pipeline allows buffering one or more batches of data ahead. This becomes important when the data processing time between batches could vary a lot. Default prefetch depth is 2. The user can change this value using the ``prefetch_queue_depth`` pipeline argument. For example, if the variation is bigger then it is recommended to prefetch more ahead.
+When the variation between batches is high, this functionality averages the processing time between
+batches when the variation between batches is high.
```

Comment:
This paragraph is not necessary - it's rephrased below.

Reply:
Done